File: kora-java-grpc-client/src/main/java/ru/tinkoff/kora/example/grpc/client/MyClientInterceptor.java
package ru.tinkoff.kora.example.grpc.client;

import io.grpc.*;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import ru.tinkoff.kora.common.Component;
import ru.tinkoff.kora.common.Tag;
import ru.tinkoff.kora.generated.grpc.UserServiceGrpc;

@Tag(UserServiceGrpc.class)
@Component
public final class MyClientInterceptor implements ClientInterceptor {

    private final Logger logger = LoggerFactory.getLogger(MyClientInterceptor.class);

    @Override
    public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
        logger.info("INTERCEPTED");
        return next.newCall(method, callOptions);
    }
}

File: examples/kotlin/kora-kotlin-grpc-client/src/main/kotlin/ru/tinkoff/kora/kotlin/example/grpc/client/MyClientInterceptor.kt
package ru.tinkoff.kora.kotlin.example.grpc.client

import io.grpc.*
import org.slf4j.LoggerFactory
import ru.tinkoff.kora.common.Component
import ru.tinkoff.kora.common.Tag
import ru.tinkoff.kora.common.annotation.Root
import ru.tinkoff.kora.generated.grpc.UserServiceGrpc

@Tag(UserServiceGrpc::class)
@Component
class MyClientInterceptor : ClientInterceptor {
    private val logger = LoggerFactory.getLogger(MyClientInterceptor::class.java)

    override fun <ReqT : Any, RespT : Any> interceptCall(
        method: MethodDescriptor<ReqT, RespT>,
        callOptions: CallOptions,
        next: Channel
    ): ClientCall<ReqT, RespT> {
        logger.info("INTERCEPTED")
        return next.newCall(method, callOptions)
    }
}

